configuration/configuration.go

```diff
@@ -0,0 +1,175 @@
+		dataDir = getDefaultArduinoDataDir()
+	}
+	sketchbookDir := viper.GetString("directories.Sketchbook")
+	if sketchbookDir == "" {
+		sketchbookDir = getDefaultSketchbookDir()
+	}
+
+	// Set default values for all the settings
+	setDefaults(dataDir, sketchbookDir)
+
+	// Attempt to read config file
+	if err := viper.ReadInConfig(); err != nil {
+		// ConfigFileNotFoundError is acceptable, anything else
+		// should be reported to the user
+		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+			feedback.Errorf("Error reading config file: %v", err)
+		}
+	}
+}
+
+// getDefaultArduinoDataDir returns the full path to the default arduino folder
+func getDefaultArduinoDataDir() string {
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		feedback.Errorf("Unable to get user home dir: %v", err)
+		return "."
+	}
+
+	switch runtime.GOOS {
+	case "linux":
+		return filepath.Join(userHomeDir, ".arduino15")
+	case "darwin":
+		return filepath.Join(userHomeDir, "Library", "Arduino15")
+	case "windows":
+		localAppDataPath, err := win32.GetLocalAppDataFolder()
+		if err != nil {
+			feedback.Errorf("Unable to get Local App Data Folder: %v", err)
+			return "."
+		}
+		return filepath.Join(localAppDataPath, "Arduino15")
+	default:
+		return "."
+	}
+}
+
+// getDefaultSketchbookDir returns the full path to the default sketchbook folder
+func getDefaultSketchbookDir() string {
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		feedback.Errorf("Unable to get user home dir: %v", err)
+		return "."
+	}
+
+	switch runtime.GOOS {
+	case "linux":
+		return filepath.Join(userHomeDir, "Arduino")
+	case "darwin":
+		return filepath.Join(userHomeDir, "Documents", "Arduino")
+	case "windows":
+		documentsPath, err := win32.GetDocumentsFolder()
+		if err != nil {
+			feedback.Errorf("Unable to get Documents Folder: %v", err)
+			return "."
+		}
+		return filepath.Join(documentsPath, "Arduino")
+	default:
+		return "."
+	}
+}
+
+// IsBundledInDesktopIDE returns true if the CLI is bundled with the Arduino IDE.
+func IsBundledInDesktopIDE() bool {
+	// value is cached the first time we run the check
+	if viper.IsSet("IDE.Bundled") {
+		return viper.GetBool("IDE.Bundled")
+	}
+
+	viper.Set("IDE.Bundled", false)
+	viper.Set("IDE.Portable", false)
+
+	logrus.Info("Checking if CLI is Bundled into the IDE")
+	executable, err := os.Executable()
+	if err != nil {
+		feedback.Errorf("Cannot get executable path: %v", err)
+		return viper.GetBool("IDE.Bundled")
+	}
+
+	executablePath, err := filepath.EvalSymlinks(executable)
+	if err != nil {
+		feedback.Errorf("Cannot get executable path: %v", err)
+		return viper.GetBool("IDE.Bundled")
+	}
+
+	ideDir := filepath.Dir(executablePath)
+
+	// We check an arbitrary number of folders that are part of the IDE
+	// install tree
+	tests := []string{
+		"tools-builder",
+		"examples/01.Basics/Blink",
+		"portable",
+	}
+
+	for _, test := range tests {
+		if _, err := os.Stat(filepath.Join(ideDir, test)); err != nil {
```

Comment:
If the check for `portable` directory fails then the IDE is set as not-bundled, but in this case the cli may still be bundled with a IDE. I think that the check for "portable" should be done separately after this one.

Reply:
Fixed